### DISCUSSION TITLE
How to avoid extra <p> tags in extensions output?

### DISCUSSION THREAD

**Discussion by khomutovspace** [datenstrom/community#473] [Ask a question] [answered]
Developing extensions I see some troubles, when empty `<p></p>` tags are likely absolutely everywhere. I guess why it can work lie this. But i can't find a solution. What is the possible way to solve it?

**Comment by schulle4u** [comment, marked as answer]
Can you please post some code examples? It will help us to better understand your problem. 

See also datenstrom/community#392 for a general explanation of shortcut output, I think this is what you're  looking for.

**Comment by BsNoSi** [comment]
`[extension …]` output is typically added *inline* except the extension is placed standalone in a line at position 1:


Markdown:
~~~
Show me [extension …] something. 
~~~
Generated code:
~~~
<p>Show me [extension output] something.</p>
~~~
Markdown:
~~~
Show me
[extension …]
something.
~~~
Generated code:
~~~
<p>Show me</p>
[extension output]
<p>something.</p>
~~~
